### PR TITLE
Adding Redlining Tools, GraphicPool and Styler

### PR DESCRIPTION
**Pull request by weskamm**
This PR adds several tools used for redlining.
We now have a container to draw points, lines and polygons.
These Features can be manipulated in several ways. There is an additional "Styler" to modify the style of the features, including an "GraphicPool" component to upload, select and delete images for point-styles.

### CODE REVIEW

File: src/view/container/RedlineStyler.js [hunk 1]
```diff
@@ -0,0 +1,627 @@
...
+                 pointStyle.fillopacity || pointStyle.strokewidth ||
+                 pointStyle.strokecolor ? new ol.style.Circle({
+                  radius: pointStyle.radius || oldStyle.getImage().getRadius(),
+                  fill: pointStyle.fillcolor || pointStyle.fillopacity ?
+                      new ol.style.Fill({
+                      color: pointStyle.fillcolor ? pointStyle.fillcolor :
+                          oldStyle.getImage().getFill().getColor(),
+                      opacity: pointStyle.fillopacity ? pointStyle.fillopacity :
+                          oldStyle.getImage().getFill().getColor()}) :
+                              oldStyle.getImage().getFill(),
```
Comment (KaiVolland) [inline src/view/container/RedlineStyler.js]: This is kind of hard to read. Nohting to be changed immediately, but just a note as this structure follows several times. Any chance to outsource this to a function?

File: src/view/window/GraphicPool.js [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+/**
+ *
+ * BasiGX.view.window.GraphicPool
+ *
+ * Used to upload, select and delete graphics
+ *
+ * @class BasiGX.view.window.GraphicPool
+ */
+Ext.define("BasiGX.view.window.GraphicPool", {
+    extend: "Ext.window.Window",
```
Comment (KaiVolland) [inline src/view/window/GraphicPool.js]: I'm personally not a friend of defining windows as classes, as you can not integrate them in any container. I'd prefer a container which can be the item of the window you want to show. This way you could add the graphicspool to any panel or so.

**Comment by KaiVolland** [comment]
Nice Gadget. Tons of codelines. But the code seems to be straight forward. Just added to minor issues. Feel free to merge anyhow.

**Comment by weskamm** [comment]
I adressed your concerns. Style creation now happens in own functions, the graphic pool has been downgraded to a panel. A container would not be sufficient, as its missing some layout things like fbar, bbar etc.

**Comment by KaiVolland** [comment]
Cool. Merge it.

**Pull request merged**